src/readsums.c

```diff
@@ -56,18 +56,25 @@ static rs_result rs_loadsig_add_sum(rs_job_t *job, rs_strong_sum_t *strong)
     rs_signature_t      *sig = job->signature;
     rs_block_sig_t      *asignature;
 
-    sig->count++;
-    new_size = sig->count * sizeof(rs_block_sig_t);
+    if (sig->count <= job->stats.sig_blocks) {
```

Comment:
It seems strange to look at `stats` here which is meant to be just a side effect.  What are you trying to check?